src/odemis/util/test/linalg_test.py

```diff
@@ -128,5 +129,82 @@ def test_are_collinear(self):
         self.assertFalse(are_collinear(non_linear_points[0], non_linear_points[1], non_linear_points[2]))
 
 
+class GenerateTriangulationTestCase(unittest.TestCase):
+    def test_max_dist_generate_triangulation_points(self):
+        """
+        Test the number of focus points when the maximum distance between focus points changes
+        """
+        # 4x4 focus points
+        max_dis = 1e-06
+        given_area_coords = (0,  0, 3.9e-06,  3.9e-06)
+        focus_points = generate_triangulation_points(max_dis, given_area_coords)
+        self.assertEqual(len(focus_points), 16)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+        # 3x3 focus points
+        max_dis = 1.5e-06  #
+        given_area_coords = (0,  0, 3.9e-06,  3.9e-06)
+        focus_points = generate_triangulation_points(max_dis, given_area_coords)
+        self.assertEqual(len(focus_points), 9)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+        # 2x2 focus points
+        max_dis = 2e-06  #
+        given_area_coords = (0,  0, 3.9e-06,  3.9e-06)
+        focus_points = generate_triangulation_points(max_dis, given_area_coords)
+        self.assertEqual(len(focus_points), 4)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+        # 1 focus point
+        max_dis = 4e-06  #
+        given_area_coords = (0,  0, 3.9e-06,  3.9e-06)
+        focus_points = generate_triangulation_points(max_dis, given_area_coords)
+        self.assertEqual(len(focus_points), 1)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+    def test_area_generate_triangulation_points(self):
+        """
+        Test the number of focus points when the given area changes
+        """
+        # 4x4 focus points when given area is a square
+        max_dis = 1e-06
+        area_coords = (0,  0, 3.9e-06,  3.9e-06)
+        focus_points = generate_triangulation_points(max_dis, area_coords)
+        self.assertEqual(len(focus_points), 16)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+        # 4x3 focus points when the given area is a rectangle
+        max_dis = 1e-06  #
+        area_coords = (0,  0, 3.9e-06,  2.9e-06)
+        focus_points = generate_triangulation_points(max_dis, area_coords)
+        self.assertEqual(len(focus_points), 12)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+    def test_focus_values_generate_triangulation_points(self):
+        """
+        Test that values of focus points are within the given area and check boundary conditions
+        """
+        # 4x4 focus points when given area is a square
+        max_dis = 1e-06
+        area_coords = (0,  0, 3.9e-06,  3.9e-06)
+        focus_points = generate_triangulation_points(max_dis, area_coords)
+        x_points = [i[0] for i in focus_points]
+        y_points = [i[1] for i in focus_points]
+        xmin = min(x_points)
+        ymin = min(y_points)
+        xmax = max(x_points)
+        ymax = max(y_points)
```

Comment:
Another way of doing this is:

```suggestion
        xmin, ymin = numpy.min(focus_points, axis=0)
        xmax, ymax = numpy.max(focus_points, axis=0)
```

pick whichever one you like best